src/libraries/illink-oob.targets

```diff
@@ -1,63 +1,78 @@
-      <_OOBReferencesFileName Include="@(_NetCoreAppRuntimeAssembliesToFileName)" Exclude="@(_OOBsToTrimFileName)" />
-      <_OOBsToTrim Include="@(_OOBsToTrimFileName -> '%(OriginalIdentity)')" />
-      <_OOBReferences Include="@(_OOBReferencesFileName -> '%(OriginalIdentity)')" />
-      <_OOBReferences Include="$(SystemPrivateCoreLibPath)" />
+      <OOBAssemblyToIgnore Include="System.CodeDom" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <OOBAssemblyToIgnore Include="System.Composition.AttributedModel" />
+      <OOBAssemblyToIgnore Include="System.Composition.Convention" />
+      <OOBAssemblyToIgnore Include="System.Composition.Hosting" />
+      <OOBAssemblyToIgnore Include="System.Composition.Runtime" />
+      <OOBAssemblyToIgnore Include="System.Composition.TypedParts" />
+      <OOBAssemblyToIgnore Include="System.Configuration.ConfigurationManager" />
+      <OOBAssemblyToIgnore Include="System.Speech" />
+
+      <NetCoreAppCurrentAssembly Include="$(NetCoreAppCurrentRuntimePath)*.dll"
+                                 Exclude="$(NetCoreAppCurrentRuntimePath)*.Generator.dll;
+                                          $(NetCoreAppCurrentRuntimePath)*.Native.dll;
+                                          $(NetCoreAppCurrentRuntimePath)*msquic.dll" />
+      <SharedFrameworkAssembly Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+
+      <!-- Move items to FileName so that we can subtract them -->
+      <NetCoreAppCurrentAssemblyWithFilename Include="@(NetCoreAppCurrentAssembly -> '%(FileName)')"
+                                             OriginalIdentity="%(Identity)" />
+      <SharedFrameworkAssemblyWithFilename Include="@(SharedFrameworkAssembly -> '%(FileName)')"
+                                           OriginalIdentity="%(Identity)" />
+
+      <OOBAssemblyToTrimWithFileName Include="@(NetCoreAppCurrentAssemblyWithFilename)"
+                                     Exclude="@(SharedFrameworkAssemblyWithFilename);
+                                              @(OOBAssemblyToIgnore)" />
+      <OOBAssemblyToTrim Include="@(OOBAssemblyToTrimWithFileName -> '%(OriginalIdentity)')" />
+      
+      <OOBAssemblyReferenceWithFilename Include="@(NetCoreAppCurrentAssemblyWithFilename)"
+                                        Exclude="@(OOBAssemblyToTrimWithFileName)" />
+      <OOBAssemblyReference Include="@(OOBAssemblyReferenceWithFilename -> '%(OriginalIdentity)')" />
+      <OOBAssemblyReference Include="$(SystemPrivateCoreLibPath)" />
     </ItemGroup>
 
     <ItemGroup>
-      <!-- Include suppression XML files bin-placed in earlier per-library linker run. -->
-      <_OOBSuppressionsXmls Include="$(ILLinkTrimAssemblyOOBSuppressionsXmlsDir)*.xml" />
+      <OOBLibraryTrimmed Include="$(OOBAssembliesTrimmedArtifactsPath)*.*" />
```

Comment:
NIT: Should this item be plural?

Reply:
Oh, I see that all across you are using singular, which I guess can make sense.

Reply:
MSBuild items are usually named singular, i.e. `Compile` and many others.